lib/internal/bootstrap/pre_execution.js

```diff
@@ -48,11 +50,49 @@ function prepareMainThreadExecution() {
   loadPreloadModules();
 }
 
+function patchProcessObject() {
+  Object.defineProperty(process, 'argv0', {
+    enumerable: true,
+    configurable: false,
+    value: process.argv[0]
+  });
+  process.argv[0] = process.execPath;
+
+  // TODO(joyeecheung): most of these should be deprecated and removed,
+  // execpt some that we need to be able to mutate during run time.
+  defineReadOnlyAlias('_eval', '--eval');
+  defineReadOnlyAlias('_print_eval', '--print');
+  defineReadOnlyAlias('_syntax_check_only', '--check');
+  defineReadOnlyAlias('_forceRepl', '--interactive');
+  defineReadOnlyAlias('_preload_modules', '--require');
+  defineReadOnlyAlias('noDeprecation', '--no-deprecation');
+  defineReadOnlyAlias('noProcessWarnings', '--no-warnings');
+  defineReadOnlyAlias('traceProcessWarnings', '--trace-warnings');
+  defineReadOnlyAlias('throwDeprecation', '--throw-deprecation');
+  defineReadOnlyAlias('profProcess', '--prof-process');
+  defineReadOnlyAlias('traceDeprecation', '--trace-deprecation');
+  defineReadOnlyAlias('_breakFirstLine', '--inspect-brk', false);
+  defineReadOnlyAlias('_breakNodeFirstLine', '--inspect-brk-node', false);
+}
+
+function defineReadOnlyAlias(name, option, enumerable = true) {
```

Comment:
Nit: I suggest to rename this to e.g., `defineProcessReadOnlyProperty`.

Reply:
I renamed the helper to `addReadOnlyProcessAlias` to avoid wrapping the longest call.